### PR TITLE
Httpclient non blocking zio files

### PR DESCRIPTION
**Pull request by ghostbuster91**
Before submitting pull request:
- [ ] Check if the project compiles by running `sbt compile`
- [ ] Verify docs compilation by running `sbt compileDocs`
- [ ] Check if tests pass by running `sbt test`
- [ ] Format code by running `sbt scalafmt`

We will have to wait until zio-nio will be available for dotty. The support was already added, so it is a matter of releasing a new version.

### CODE REVIEW

File: httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioBodyFromHttpClient.scala [hunk 1]
```diff
@@ -1,77 +1,112 @@
...
-          response: WebSocket[BlockingTask],
+          response: WebSocket[Task],
           e: GotAWebSocketException
-      ): BlockingTask[Unit] = response.close()
+      ): Task[Unit] = response.close()
     }
 
-  override implicit def monad: MonadError[BlockingTask] = new RIOMonadAsyncError[Blocking]
+  private def readAllBytes(fileChannel: AsynchronousFileChannel) = {
+    val bufferSize = 4096
```
Comment (ghostbuster91) [inline httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioBodyFromHttpClient.scala:100]: Should we restore chunkSize parameter in constructor and pass this value all the way down to here?

Comment (adamw) [reply httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioBodyFromHttpClient.scala:100]: I can't imagine really a situation where you'd want to customise this. Let's leave it, 4K seems like a good magic number ;)

File: docs/backends/summary.md [hunk 2]
```diff
@@ -19,27 +19,28 @@ Each backend has two type parameters:
...
+``OkHttpFutureBackend``              ``scala.concurrent.Future``  n/a                                               yes (regular)              no
+``OkHttpMonixBackend``               ``monix.eval.Task``          ``monix.reactive.Observable[ByteBuffer]``         yes (regular & streaming)  no
+``Http4sBackend``                    ``F[_]: cats.effect.Effect`` ``fs2.Stream[F, Byte]``                           no                         no
+``HttpClientSyncBackend``            None (``Identity``)          n/a                                               no                         no
+``HttpClientFutureBackend``          ``scala.concurrent.Future``  n/a                                               yes (regular)              no
+``HttpClientMonixBackend``           ``monix.eval.Task``          ``monix.reactive.Observable[ByteBuffer]``         yes (regular & streaming)  no
+``HttpClientFs2Backend``             ``F[_]: cats.effect.Async``  ``fs2.Stream[F, Byte]``                           yes (regular & streaming)  yes
+``HttpClientZioBackend``             ``zio.RIO[Blocking, *]``     ``zio.stream.ZStream[Blocking, Throwable, Byte]`` yes (regular & streaming)  yes
```
Comment (adamw) [inline docs/backends/summary.md:42]: this shouldn't have Blocking, right?

**Comment by adamw** [comment]
Released in [3.0.0-RC5](https://github.com/softwaremill/sttp/releases/tag/v3.0.0-RC5), thanks!

**Comment by ghostbuster91** [comment]
@adamw But it doesn't compile on dotty, as I mentioned :)

**Comment by ghostbuster91** [comment]
Ok, now I see that there is a magic flag to control it :)

**Comment by adamw** [comment]
Yes, when https://github.com/zio/zio-nio/issues/182 is released we'll re-enable

**Comment by jczuchnowski** [comment]
ZIO-NIO 1.0.0-RC10 has dotty 0.27 support.

**Pull request merged**